### PR TITLE
handle OVR Platform entitlement checks

### PR DESCRIPTION
**Pull request by birarda**
- adds entitlement checks for Oculus Store when OCULUS_APP_ID is set during build and oculus store command line argument is passed

### CODE REVIEW

**Comment by birarda** [comment]
#### Test Plan

- Run this PR build in Oculus mode. It should launch and keep running as normal.
- Change Interface.exe from this PR build to pass a command line argument `--oculus-store`
- Launch this PR and use the Oculus display mode. It should quit immediately if you launch directly in Oculus mode or should quit immediately once you switch to the Oculus display mode.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 46086cdcceb4b25a1e8cbf73e03a787f2a56fc39 available at: https://gist.github.com/d63bc6b78256f44d81a5936e023f1db8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9318-46086cd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9318-46086cd.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8f279ce1f7db4367bfce1d6362cad4768d8f0439 available at: https://gist.github.com/d63bc6b78256f44d81a5936e023f1db8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9318-8f279ce.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9318-8f279ce.dmg

File: plugins/oculus/src/OculusHelpers.cpp [hunk 1]
```diff
@@ -89,6 +93,18 @@ ovrSession acquireOculusSession() {
...
         }
 
+#ifdef OCULUS_APP_ID
+        if (qApp->property(hifi::properties::OCULUS_STORE).toBool()) {
+            if (ovr_PlatformInitializeWindows(OCULUS_APP_ID) != ovrPlatformInitialize_Success) {
+                // we were unable to initialize the platform for entitlement check - fail the check
+                _quitRequested = true;
+            } else {
+                qCDebug(oculus) << "Performing Oculus Platform entitlement check";
+                ovr_Entitlement_GetIsViewerEntitled();
```
Comment (jherico) [inline plugins/oculus/src/OculusHelpers.cpp]: do we not need to check the return value here?  It seems like one of the tests Oculus might run is verifying that if the entitlement isn't available that the app shuts down.

Comment (jherico) [inline plugins/oculus/src/OculusHelpers.cpp]: nm, I see that it's asynchronous.

**Comment by OmegaHeron** [comment]
Tested Windows 10
Launched from stock shortcut into desktop mode.  Pass
Switched to DK2 HMD from ^^.  Pass
Switched back to Desktop from ^^.  Pass

Launched from stock shortcut into DK2 mode.  Pass
Switched to Desktop mode from ^^. Pass
Switched back to DK2 mode from ^^. Pass

Launched this PR with command flag --oculus-store in desktop mode.  Switched to DK2 mode - it switched to DK2 and continued working.

Launched this PR with command flag --oculus-store in DK2 mode, interface fired up and worked as expected.

If I'm understanding this PR correctly that's a QA Fail.  I confirmed I see build PR9318 in title bar, I launched it from command prompt with `interface.exe --oculus-store ` from PR directory...

**Comment by birarda** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8f279ce1f7db4367bfce1d6362cad4768d8f0439 available at: https://gist.github.com/d63bc6b78256f44d81a5936e023f1db8<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9318-8f279ce.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9318-8f279ce.dmg

**Comment by birarda** [comment]
@OmegaHeron the build before did not have an env variable set that was required to enable the behaviour I was having you test. Can you please test using the same test plan again?

**Comment by OmegaHeron** [comment]
Will do - be a bit, am out of work space for a short time.

**Comment by OmegaHeron** [comment]
QA Pass Windows 10

Tested with DK2
If running with --oculus-store flag Interface terminates on launch if in HMD mode or after being switched to HMD mode from Desktop mode.  Without flag, Interface works as it always has.

**Pull request merged**